apis/vpcresources/v1beta1/vpcresourcerequest_types.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// VPCResourceRequestSpec defines the desired state of VPCResourceRequest
+type VPCResourceRequestSpec struct {
+	RequestTypes  map[string]ResourceType `json:"requestTypes,omitempty"`
+	RequestTarget Target                  `json:"requestTarget,omitempty"`
+	Message       string                  `json:"message,omitempty"`
+}
+
+// Target represents a request's target
+// Target could be pod, node or other Kinds
+type Target struct {
+	Name  string `json:"name,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+// Type represents a request's type
+// ResourceType could features, such as SGP, custom networking, etc.
+type ResourceType struct {
+	Name  string `json:"name,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:printcolumn:name="Target",type=string,JSONPath=`.spec.requestTarget`,description="The target of the request CRD object"
+// +kubebuilder:printcolumn:name="Types",type=string,JSONPath=`.spec.requestTypes`,description="The type of the request CRD object"
+// +kubebuilder:printcolumn:name="Message",type=string,JSONPath=`.spec.message`,description="The message of the request CRD object"
+// +kubebuilder:resource:shortName=vrr
+
+// Custom Resource Definition for Request to VPC resource controller
+type VPCResourceRequest struct {
```

Comment:
Thoughts on "VPCResource"? Request is implicit -- essentially all kubernetes apis are requesting that something happens.

Reply:
This was suggested to indicate something requested to VPC resource controller to manage. Probably we should be specific by using "VPCResourceController"?